File: quiz_application_front/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';
import { Observable } from 'rxjs';
import { Question } from '../Classes/question';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http: HttpClient) { }
  public getQuestionOfQuiz(qId : any){
    return this._http.get(`${baseUrl}/question/quiz/all/${qId}`);

  }

   //add question

   public addQuestion(question :Question):Observable<object>{
    return this._http.post(`${baseUrl}/question/`,question);

  }

   //delete question
   public deleteQuestion(questionId: Number){
    return this._http.delete(`${baseUrl}/question/${questionId}`);

  }

  //
  public getQuestionById(quesId:number):Observable<Question>{
    return this._http.get<Question>(`${baseUrl}/question/${quesId}`);
  }

  public updateQuestion( quesId:number, question :Question):Observable<object>{
    console.log('inside services')
    console.log(quesId);
    console.log(question);
    return this._http.put(`${baseUrl}/question/${quesId}`,question);

  }


  public getQuestionOfQuizForTest(qId :number):Observable<Question[]>{
    return this._http.get<Question[]>(`${baseUrl}/question/quiz/${qId}`);

  }

  //eval-quiz
  public evalQuiz(questions:Question[]):Observable<Object>{
    return this._http.post(`${baseUrl}/question/eval-quiz`,questions)
  }
}

File: quiz_application_Back/src/main/java/com/quiz/service/QuestionService.java
package com.quiz.service;

import java.util.Set;

import com.quiz.dto.QuestionsDto;
import com.quiz.dto.QuizDto;
import com.quiz.entity.Questions;
import com.quiz.entity.Quiz;

public interface QuestionService {
	
	public QuestionsDto addQuestion(QuestionsDto questionsDto);
	
	Set<QuestionsDto> getAllQuestions();
	
	public  QuestionsDto updateQuestion(QuestionsDto questionDto,long quesId);
	
	public  QuestionsDto getQuestion(Long questionId);
	
	 public void  deleteQuestion(Long quesId);
	 
	 public  Set<QuestionsDto> getQuestionOfQuiz(QuizDto quizDto);
	 public Questions get(Long questionId);

}
